lib/gala/payment_token.rb

```diff
@@ -37,6 +39,30 @@ def decrypt(certificate_pem, private_key_pem)
 
     class << self
 
+      def validate_signature(signature, ephemeral_public_key, data, transaction_id, application_data)
+        leaf_cert = nil
+        intermediate_cert = nil
+        p7 = OpenSSL::PKCS7.new(Base64.decode64(signature))
+        p7.certificates.each {|c|
+          c.extensions.each { |e|
+            leaf_cert = c if e.oid == '1.2.840.113635.100.6.29'
+            intermediate_cert = c if e.oid == '1.2.840.113635.100.6.2.14'
+          }
+        }
+        raise InvalidSignatureError unless leaf_cert && intermediate_cert
+        raise InvalidSignatureError unless leaf_cert.issuer == intermediate_cert.subject
+        expected_issuer = OpenSSL::X509::Name.parse "/CN=Apple Root CA - G3/OU=Apple Certification Authority/O=Apple Inc./C=US"
```

Comment:
I have been unable to use the cert referenced in the docs.  I am going to try to use Java to generate a pem file this week so that the whole process is complete according to the docs.  This line simply hacks in the name of the root cert.  I would feel better with actually comparing the issuer/subject of the root cert with the issuer of the intermediate cert.